### DISCUSSION TITLE
Writing gif to MemoryStream only saves the first frame

### DISCUSSION THREAD

**Discussion by sywor** [dlemstra/Magick.NET#863] [Help]
Hi!
I'm trying the generate gifs and saving them to minio by using a MemoryStream to hand over the data to the minio WriteAsync method. This works just fine for all other files, even Magic.NET generated png files. But not gifs!? 

This is the method I use. 

```csharp
private async Task<MinioFile> GenerateGif(IEnumerable<MagickImage> _images)
{
        logger.LogInformation("Generating new gif");

        using var gif = new MagickImageCollection();

        foreach (var image in _images)
        {
            gif.Add(image);
            gif.Last().AnimationDelay = 5;
        }

        gif.Last().AnimationDelay = 50;

        gif.Optimize();
        gif.OptimizeTransparency();

        var fileName = $"{Guid.NewGuid()}.gif";

        logger.LogInformation("Saving");

        var saveStream = new MemoryStream();
        await gif.WriteAsync(saveStream, MagickFormat.Gif);
        var minioFile = MinioFile.Of("gif", fileName);
        await minioService.PutEphemeralObjectAsync(minioFile, saveStream, "image/gif");

        logger.LogInformation("Saving done");

        return minioFile;
}
```

When I replace the 
```csharp
await gif.WriteAsync(saveStream, MagickFormat.Gif);
```
With:
```csharp
gif.Write("debug.gif", MagickFormat.Gif);
```
It works just fine!
I'm I using the MemoryStream wrong? How else can you move data from one method that writes to a stream to another that reads from a stream?

**Comment by dlemstra** [comment]
Are you sure this is not an issue in the Minio library? I just made the following change to one of my unit tests: https://github.com/dlemstra/Magick.NET/commit/6b043e551db9336bf5e5ba44c0e457434411254d. And this unit test is passing. And when I change the output format in that test to GIF instead of TIFF it still passes.

**Comment by sywor** [comment]
I'm fairly sure it has nothing to do with minio for one reason. I just tried to write to FileStream instead:
```csharp
var tempFileName = Path.GetTempFileName();
await using var saveStream = new FileStream(tempFileName, FileMode.Open, FileAccess.ReadWrite);
await gif.WriteAsync(saveStream);

saveStream.Flush();
saveStream.Close();
```
And the resulting file is only 68KB and does not animate. 
When i did :
```csharp
gif.Write("debug.gif", MagickFormat.Gif); 
```
directly the resulting file was close to 1MB in size.

**Reply by dlemstra** [reply]
You are not comparing the same thing. `await gif.WriteAsync(saveStream)` does not always do the same as `await gif.WriteAsync(saveStream, MagickFormat.Gif)`. If you have a small sample that demonstrates this issue I can investigate further.

**Reply by sywor** [reply]
My fault. I thought I used the WriteAsync with the second MagicFormat argument but I went back and checked it turns out I used the WriteAsync with only the stream argument.
